### DISCUSSION TITLE
Question about BATCH method in PocketBase SDK (Dart/Flutter)

### DISCUSSION THREAD

**Discussion by EssekerDev** [pocketbase/pocketbase#5994] [Q&A] [answered]
Hello,

I have a question about the batch method in PocketBase. Is it possible to retrieve an ID from a previous operation within the same batch?

_For example:_

1. The first batch operation creates an entry in Collection A.

2. The second operation creates an entry in Collection B but needs to reference the ID of the entry created in Collection A.

3. After creating the entry in Collection B, the last operation updates the entry in Collection A to include the ID of the entry in Collection B in its relation field.

I’m using the PocketBase SDK for Dart/Flutter. Is there a way to achieve this workflow with the batch method? If not, what would be the best approach to handle this scenario?

Thanks in advance for your help!

**Comment by ganigeorgiev** [comment, marked as answer]
There is no option at the moment to retrieve the autogenerated id before the entire batch operation completes.

An alternative could be to submit a custom id with the body payload and use that id for reference in the relation  field(s). Note that the id constraints can be customized from the related collection field options but by default is restricted to 15 chars.

For example:

```dart
final batch = pb.createBatch();

batch.collection('collectionA').create(body: { "id": "testA", ... });
batch.collection('collectionB').create(body: { "yourRelField": "testA", ... });

final result = await batch.send();
```

**Reply by EssekerDev** [reply]
Thank you very much for your detailed response! This approach with custom IDs makes a lot of sense, and I’ll give it a try. I appreciate your help and the clarification about customizing the ID constraints.
